### PR TITLE
Migrate core/shortcut_registry.js to goog.module syntax

### PR DESCRIPTION
**Pull request by gonfunko**
## The basics

- [x] I branched from `goog_module`
- [x] My pull request is against `goog_module`
- [x] My code follows the [style guide](
      https://developers.google.com/blockly/guides/modify/web/style-guide)
- [x] My code is presented in the form suggested in the [module
      conversion guide](https://github.com/google/blockly/issues/5026)
- [x] I have run `npm test`.

### Resolves

Part of #5026

### Proposed Changes

Converts `core/shortcut_registry.js` to `goog.module` with ES6 `const`/`let`.

### CODE REVIEW

**Review by BeksOmega — APPROVED** [review]
LGTM! But I'm not sure if we need to make any changes wrt how singleton instances are handled. I think we should get @cpcallen 's opinion before merging.

File: core/interfaces/i_keyboard_accessible.js
Comment (cpcallen) [inline core/interfaces/i_keyboard_accessible.js:18]: This de-destructuring probably wasn't *really* necessary: since this is a `requireType` (and not just `require`) it can't be necessary to stub/mock this dependency in tests—and the name `KeyboardShortcut` seems adequately unambiguous on its own.

File: core/shortcut_registry.js
Comment (cpcallen) [inline core/shortcut_registry.js:31]: Having the constructor overwrite the singleton seems like a bad idea (especially given the constructor is exported!)  No fix required now, but see below.

File: core/interfaces/i_keyboard_accessible.js
Comment (cpcallen) [inline core/interfaces/i_keyboard_accessible.js:18]: This de-destructuring probably wasn't *really* necessary: since this is a `requireType` (and not just `require`) it can't be necessary to stub/mock this dependency in tests—and the name `KeyboardShortcut` seems adequately unambiguous on its own.

File: core/shortcut_registry.js
Comment (cpcallen) [inline core/shortcut_registry.js:353]: Fine for now (to keep API compatibility until the first pass is finished), but it should probably be changed to something like:
```JS
exports.shortcutRegistry = new ShortcutRegistry();
```
i.e., only exporting the singleton, not the constructor (and not having the constructor overwrite any previously-created singleton).

Added to the list in #5073.

**Review by cpcallen — APPROVED** [review]
LGTM, but see comments below re: future work.

File: core/interfaces/i_keyboard_accessible.js
Comment (gonfunko) [reply core/interfaces/i_keyboard_accessible.js:18]: The compiler yelled without it, I believe about not being able to destructure a default export.

File: core/shortcut_registry.js
Comment (gonfunko) [reply core/shortcut_registry.js:353]: Yeah that does seem much nicer, good call!

File: core/interfaces/i_keyboard_accessible.js
Comment (cpcallen) [reply core/interfaces/i_keyboard_accessible.js:18]: Oh, yeah.  Right.  I forgot it has _opinions_ about this sort of thing.

Moving away from default exports will help solve this.

**Pull request merged**
